apollo-router/src/query_planner/fetch.rs

```diff
@@ -364,13 +660,23 @@ impl FetchNode {
             parameters.supergraph_request,
             parameters.schema,
             &self.input_rewrites,
+            &self.context_rewrites,
         ) {
             Some(variables) => variables,
             None => {
                 return (Value::Object(Object::default()), Vec::new());
             }
         };
 
+        let query_batched_query = if let Ok(qbq) =
+            query_batching_for_contextual_args(operation.as_serialized(), &contextual_args)
```

Comment:
There's no point making us re-parse the operation by passing it as a string.

```suggestion
            query_batching_for_contextual_args(operation.as_parsed(), &contextual_args)
```